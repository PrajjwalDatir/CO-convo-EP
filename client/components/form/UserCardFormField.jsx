import React from 'react';
import PropTypes from 'prop-types';
import { useFormContext } from 'react-hook-form';
import Input from '#/components/input';

function UserCardFormField({ label, name, type, register }) {
  const { errors } = useFormContext();

  return (
    <div>
      <label htmlFor={`${name}-field`}>
        <div className="flex">
          <div className="mb-1 uppercase font-bold tracking-wide text-xs text-gray-700">
            {label}
          </div>
          <div className="text-xs text-red-700 ml-auto" id={`${name}-field-error`}>
            {errors[name] && errors[name].message}
          </div>
        </div>
        <Input
          className="w-full mb-4"
          id={`${name}-field`}
          name={name}
          type={type}
          aira-invalid={errors[name] ? 'true' : 'false'}
          aira-describedby={`${name}-field-error`}
          ref={register}
        />
      </label>
    </div>
  );
}

UserCardFormField.propTypes = {
  label: PropTypes.node.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string,
  register: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({ current: PropTypes.any }),
  ]).isRequired,
};

UserCardFormField.defaultProps = {
  type: 'text',
};

export default UserCardFormField;
