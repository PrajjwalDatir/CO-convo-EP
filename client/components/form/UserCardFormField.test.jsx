import React from 'react';
import { mount } from 'enzyme';
import { FormContext } from 'react-hook-form';
import * as R from 'ramda';
import UserCardFormField from './UserCardFormField';

describe('<UserCardFormField />', () => {
  it('should match snapshot', () => {
    const wrapper = mount(
      <FormContext {...{ errors: jest.fn() }}>
        <UserCardFormField
          label="Password"
          name="password"
          type="password"
          register={R.always()}
        />
        ,
      </FormContext>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
