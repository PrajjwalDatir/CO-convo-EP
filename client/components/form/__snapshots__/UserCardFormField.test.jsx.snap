// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCardFormField /> should match snapshot 1`] = `
<FormContext
  errors={[MockFunction]}
>
  <UserCardFormField
    label="Password"
    name="password"
    register={[Function]}
    type="password"
  >
    <div>
      <label
        htmlFor="password-field"
      >
        <div
          className="flex"
        >
          <div
            className="mb-1 uppercase font-bold tracking-wide text-xs text-gray-700"
          >
            Password
          </div>
          <div
            className="text-xs text-red-700 ml-auto"
            id="password-field-error"
          />
        </div>
        <ForwardRef
          aira-describedby="password-field-error"
          aira-invalid="false"
          className="w-full mb-4"
          id="password-field"
          name="password"
          type="password"
        >
          <input
            aira-describedby="password-field-error"
            aira-invalid="false"
            className="rounded-lg bg-gray-200 px-3 py-2 border border-solid border-transparent outline-none hover:border-gray-400 focus:bg-white focus:border-gray-400 w-full mb-4"
            id="password-field"
            name="password"
            type="password"
          />
        </ForwardRef>
      </label>
    </div>
  </UserCardFormField>
  ,
</FormContext>
`;
